xdsl/traits.py

```diff
@@ -94,6 +94,69 @@ def verify(self, op: Operation) -> None:
                 )
 
 
+class SingleBlockImplicitTerminator(OpTrait):
+    """
+    Checks and adds if missing the specified terminator to an operation which has
+    single-block region.
+
+    This should be fully compatible with MLIR's Trait:
+    https://mlir.llvm.org/docs/Traits/#single-block-with-implicit-terminator
+    """
+
+    parameters: type[Operation]
+
+    def __init__(self, parameters: type[Operation]):
```

Comment:
Is this necessary?

Reply:
which part? the ctor or the parameter?

Reply:
I forgot it was a `dataclass`, removed the ctor.